pkg/sdk/client.go

```diff
@@ -300,6 +301,19 @@ func (c *Client) queryOne(ctx context.Context, dest interface{}, sql string) err
 		log.Printf("[DEBUG] sql-conn-query-one-dry: %v\n", sql)
 		return nil
 	}
-	ctx = context.WithValue(ctx, snowflakeAccountLocatorContextKey, c.accountLocator)
+	ctx = context.WithValue(ctx, SnowflakeAccountLocatorContextKey, c.accountLocator)
+	sql = appendQueryMetadata(ctx, sql)
 	return decodeDriverError(c.db.GetContext(ctx, dest, sql))
 }
+
+func appendQueryMetadata(ctx context.Context, sql string) string {
+	if metadata, ok := tracking.FromContext(ctx); ok {
+		newSql, err := tracking.AppendMetadataToSql(sql, metadata)
+		if err != nil {
+			log.Printf("[ERROR] failed to append metadata tracking: %v\n", err)
+			return sql
+		}
+		return newSql
```

Comment:
Can we also log here if there's no metadata in the context?

Reply:
I'm wondering if that's necessary now (could be added later) because it would log this message a lot